src/test/java/com/google/sps/CommentServiceTest.java

```diff
@@ -0,0 +1,87 @@
+package com.google.sps;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import com.google.sps.servlets.CommentService;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+
+public final class CommentServiceTest {
+
+  CommentService commentService = new CommentService();
+
+  /**
+   * A list of Strings should be returned for any public video that has comments. 
+   */
+  @Test
+  public void publicVideoReturnsCommentsList() throws GeneralSecurityException, IOException {
+    String videoId = "v3whnmU619I"; // Public video that has some comments.
+    List<String> actual = commentService.getCommentsFromId(videoId);
+
+    Assert.assertFalse(actual.isEmpty());
+  }
+
+  /**
+   * Unlisted videos should be accessible to the API if the video id is inputted. List of comments
+   * should be returned for unlisted videos just like they would for a public video. 
+   */
+  @Test
+  public void unlistedVideoReturnsCommentsList() throws GeneralSecurityException, IOException {
+    String videoId = "9BUFUUhfXSg"; // Unlisted test video that has 2 comments. 
+    List<String> actual = commentService.getCommentsFromId(videoId);
+
+    Assert.assertFalse(actual.isEmpty());
+  }
+
+  /**
+   * Private videos should not be accessible with the YouTube API. In this case, a 
+   * GoogleJsonResponseException should be caught and null should be returned. 
+   */
+  @Test
+  public void privateVideoReturnsNull() throws GeneralSecurityException, IOException {
+    String videoId = "qYnSnsysgVk";
+    List<String> actual = commentService.getCommentsFromId(videoId);
+
+    Assert.assertEquals(null, actual);
+  }
+  
+  /**
+   * For videos that have disabled commenting, a GoogleJsonResponseException should be caught and
+   * null should be returned. 
+   */
+  @Test
```

Comment:
great test! I did not think of this case

Reply:
+1 I didn't even know it is possible to disable commenting 😅